webpack/assets/javascripts/react_app/components/charts/StatisticsChartBox.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import Component from '../common/BindingComponent';
+import ChartHeader from './ChartHeader';
+import Chart from './Chart';
+import ChartModal from './ChartModal';
+
+import StatisticsStore from '../../stores/StatisticsStore';
+import statisticsPage from '../../../pages/statistics_page';
+
+export default class extends Component {
+    constructor(props) {
+        super(props);
+        this.state = { showModal: false, isLoaded: false };
+        this.bind(
+            'drawChart',
+            'onChange',
+            'onClick',
+            'closeModal',
+            'openModal',
+            'drawModal'
+        );
+    }
+
+    componentDidMount() {
+        StatisticsStore.addChangeListener(this.onChange);
+    }
+
+    componentWillUnmount() {
+        StatisticsStore.removeChangeListener(this.onChange);
+    }
+
+    onChange() {
+        const statistics = StatisticsStore.getStatisticsData(this.props.id);
+
+        this.setState({ isLoaded: true, hasData: !!statistics.data.length, data: statistics.data });
```

Comment:
I'm still seeing the case where the chart data is not available yet the spinner stops, could this be that you are checking for when the promise started but not when it was actually finished?